src/main/scala/strawman/collection/mutable/ArrayDeque.scala

```diff
@@ -184,7 +184,27 @@ class ArrayDeque[A] private(var array: Array[AnyRef], var start: Int, var end: I
 
   override def clone() = new ArrayDeque(array.clone, start, end)
 
-  override def clear() = if (nonEmpty) set(array = ArrayDeque.alloc(ArrayDeque.DefaultInitialSize), start = 0, end = 0)
+  /**
+    * Note: This does not actually resize the internal representation.
+    * See clearAndShrink if you want to also resize internally
+    */
+  override def clear() = {
+    nullify()
+    start = 0
+    end = 0
+  }
```

Comment:
Cakes don't play nicely with `map` and friends.

Internal state is the way to go.